### PR TITLE
fixed bug in building initialization and cleaned up building code

### PR DESCRIPTION
**Pull request by aberke**
issues fixed are important for implementing motifs:
- buildings were not being initialized with proper R/O/A capacities
- buildings on grid were overwriting previously initialized buildings

- also broke building class out into its own file

### CODE REVIEW

**Review by agrignard — COMMENTED** [review]
When you say "buildings on grid were overwriting previously initialized buildings" 
I am not sure to understand what was the problem before? was it a documented issue?

I don't understand the replacement you did on line 37 /38 the for loop was doing the same with less line no?

File: ABMobility/Grid.pde [hunk 1]
```diff
@@ -31,11 +32,25 @@ public class Grid {
      
      table = loadTable("block/Cooper Hewitt Buildings - Building Blocks.csv", "header");
      for (TableRow row : table.rows()) {
-       buildings.add(new Building(gridMap.get(row.getInt("id")),row.getInt("id"),row.getInt("R"),row.getInt("O"),row.getInt("A")));
```
Comment (aberke) [inline ABMobility/Grid.pde:34]: @agrignard at this line the buildings were being instantiated, but they were not then being used.

File: ABMobility/Grid.pde [hunk 2]
```diff
@@ -31,11 +32,25 @@ public class Grid {
...
+      int loc = id;  // initial location is same as building id
+      int capacityR = row.getInt("R");
+      int capacityO = row.getInt("O");
+      int capacityA = row.getInt("A");
+      Building b = new Building(gridMap.get(loc), id, capacityR, capacityO, capacityA);
+      buildings.add(b);
      }
      
-     for (int i=0;i<=18;i++){
-       buildingsOnGrid.add(new Building(gridMap.get(i),i,0,0,0));
```
Comment (aberke) [inline ABMobility/Grid.pde:38]: At this line the buildings that were previously instantiated were not being used and instead buildings with dummy values were being added to the grid.
In order to dynamically choose amenity locations, we should be able to choose buildings on the grid that have greater capacity for amenities.  That information was being overwritten.

**Comment by aberke** [comment]
No it is different - I left some comments inline for clarity :)

File: ABMobility/Grid.pde (hunk 2 above)
Comment (agrignard) [reply ABMobility/Grid.pde:38]: ok I see , there was no R,0 and A (I though it was taken from the data) but yes indeed it's better to populate them with the right value

**Pull request merged**